src/resources/nfts/index.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  QueryClientConfig,
+  useInfiniteQuery,
+  useQuery,
+} from '@tanstack/react-query';
+import { createQueryKey, queryClient } from '@/react-query';
+import { NFT } from '@/resources/nfts/types';
+import { UniqueAsset } from '@/entities/uniqueAssets';
+import { fetchSimplehashNFTs } from '@/resources/nfts/simplehash';
+import { useEffect } from 'react';
+import {
+  filterSimplehashNFTs,
+  simplehashNFTToUniqueAsset,
+} from '@/resources/nfts/simplehash/utils';
+import { rainbowFetch } from '@/rainbow-fetch';
+
+const NFTS_LIMIT = 2000;
+const NFTS_STALE_TIME = 300000; // 5 minutes
+const POLYGON_ALLOWLIST_STALE_TIME = 600000; // 10 minutes
+
+export const nftsQueryKey = ({ address }: { address: string }) =>
+  createQueryKey('nfts', { address }, { persisterVersion: 1 });
+
+export const nftsStreamQueryKey = ({ address }: { address: string }) =>
+  createQueryKey('nftsStream', { address }, { persisterVersion: 1 });
+
+async function fetchPolygonAllowlist(): Promise<string[]> {
+  return await queryClient.fetchQuery(
+    ['polygon-allowlist'],
+    async () =>
+      (
+        await rainbowFetch(
+          'https://metadata.p.rainbow.me/token-list/137-allowlist.json',
+          { method: 'get' }
+        )
+      ).data.data.addresses,
+    {
+      staleTime: POLYGON_ALLOWLIST_STALE_TIME,
+    }
+  );
+}
+
+function usePolygonAllowlist() {
+  return useQuery<string[]>({
+    queryKey: ['polygon-allowlist'],
+    queryFn: async () =>
+      (
+        await rainbowFetch(
+          'https://metadata.p.rainbow.me/token-list/137-allowlist.json',
+          { method: 'get' }
+        )
+      ).data.data.addresses,
+    staleTime: POLYGON_ALLOWLIST_STALE_TIME,
+  });
+}
+
+export async function fetchLegacyNFTs({
+  address,
+  queryClientConfig,
+}: {
+  address: string;
+  queryClientConfig: QueryClientConfig;
+}): Promise<UniqueAsset[]> {
+  return queryClient.fetchQuery(
+    nftsQueryKey({ address }),
+    async () => {
+      let finished = false;
+      let cursor: string | undefined;
+      let nfts: UniqueAsset[] = [];
+      while (!finished) {
+        // eslint-disable-next-line no-await-in-loop
+        const [simplehashResponse, polygonAllowlist] = await Promise.all([
+          fetchSimplehashNFTs(address, cursor),
+          fetchPolygonAllowlist(),
+        ]);
+
+        const { data: simplehashNFTs, nextCursor } = simplehashResponse;
+
+        const newNFTs = filterSimplehashNFTs(
+          simplehashNFTs,
+          polygonAllowlist
+        ).map(simplehashNFTToUniqueAsset);
+
+        nfts = nfts.concat(newNFTs);
+
+        if (nextCursor && nfts.length < NFTS_LIMIT) {
+          cursor = nextCursor;
+        } else {
+          // eslint-disable-next-line require-atomic-updates
+          finished = true;
+        }
+      }
+      return nfts;
+    },
+    { staleTime: NFTS_STALE_TIME, ...queryClientConfig }
```

Comment:
supporting custom staleTime here for uses cases where we definitely need fresh data, such as when the user pulls to refresh their WalletScreen